drivers/sensor/lps2xdf/lps2xdf.h

```diff
@@ -0,0 +1,106 @@
+/* ST Microelectronics LPS2XDF pressure and temperature sensor
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheets:
+ * https://www.st.com/resource/en/datasheet/lps22df.pdf
+ * https://www.st.com/resource/en/datasheet/lps28dfw.pdf
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_LPS2XDF_LPS2XDF_H_
+#define ZEPHYR_DRIVERS_SENSOR_LPS2XDF_LPS2XDF_H_
+
+#include <stdint.h>
+#include <stmemsc.h>
+
+//#if DT_HAS_COMPAT_STATUS_OKAY(st_lps22df)
+#include "lps22df_reg.h"
+//#endif
+
+//#if DT_HAS_COMPAT_STATUS_OKAY(st_lps28dfw)
+#include "lps28dfw_reg.h"
+//#endif
+
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i3c.h>
+
+#if (DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps22df, i3c) || \
+	DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps28dfw, i3c))
+	#define ON_I3C_BUS(cfg) (cfg->i3c.bus != NULL)
+#else
+	#define ON_I3C_BUS(cfg) (false)
+#endif
+
+struct lps2xdf_config {
+	stmdev_ctx_t ctx;
+	union {
+#if (DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps22df, i2c) || \
+	DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps28dfw, i2c))
+		const struct i2c_dt_spec i2c;
+#endif
+#if DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps22df, spi)
+		const struct spi_dt_spec spi;
+#endif
+#if (DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps22df, i3c) || \
+	DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps28dfw, i3c))
+		struct i3c_device_desc **i3c;
+#endif
+	} stmemsc_cfg;
+	uint8_t odr;
+	uint8_t lpf;
+	uint8_t avg;
+	uint8_t drdy_pulsed;
+#ifdef CONFIG_LPS2XDF_TRIGGER
+	struct gpio_dt_spec gpio_int;
+#endif
+
+#if (DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps22df, i3c) || \
+	DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(st_lps28dfw, i3c))
+	struct {
+		const struct device *bus;
+		const struct i3c_device_id dev_id;
+	} i3c;
+#endif
+	const void *chip_api;
```

Comment:
you can use `lps2xdf_chip_api` here, it's not like you are making this generic

Reply:
Ok, changed it.